enginetest/harness.go

```diff
@@ -35,11 +40,16 @@ type Harness interface {
 	// additional information (e.g. current DB) set uniformly. To replicated the behavior of tests during setup,
 	// harnesses should generally dispatch to enginetest.NewContext(harness), rather than calling this method themselves.
 	NewContext() *sql.Context
+	// SetSetup injects a test suite's setup scripts
+	SetSetup(...[]Testdata)
```

Comment:
Also aren't `NewTable`, `NewDatabase`, `NewDatabaseProvider` etc all defunct now? If the harness is responsible for creating an engine with all the test data provided, I don't see why it would need any of these older methods

Reply:
A related comment is that you have added to this base interface instead of creating an extension to it. That's fine (it breaks existing integrators but we can live with that), but if you're goig to do it this way , then commit

Reply:
We have to keep these until someone finishes the refactor, maybe 20% of the enginetests still use these, and people keep adding more